test/test.py

```diff
@@ -394,6 +395,10 @@ def _assert_bundle(self, uuid, version, manifest, metadata_files,
         self.assertEqual(library_construction_methods, {p.library_construction_method for p in library_prep_protos})
         self.assertEqual(library_construction_methods, {p.library_construction_approach for p in library_prep_protos})
 
+        # All SequencingProtocol instances should have a required paired_end property equal to True or False
+        for sequencing_protocol in [p for p in bundle.protocols.values() if isinstance(p, SequencingProtocol)]:
```

Comment:
Nit: You could move the `if` from the comprehension into the body of the `for` loop. That would eliminate the additional temporary list and maybe be more readable.